numpy/ma/extras.py

```diff
@@ -770,23 +809,10 @@ def compress_rowcols(x, axis=None):
     x = asarray(x)
     if x.ndim != 2:
         raise NotImplementedError("compress2d works for 2D arrays only.")
```

Comment:
It's probably a good opportunity to fix the error message to match the actual function being called.